src/main/java/com/codeborne/selenide/appium/commands/AppiumScrollTo.java

```diff
@@ -0,0 +1,85 @@
+package com.codeborne.selenide.appium.commands;
+
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.commands.ScrollTo;
+import com.codeborne.selenide.impl.WebElementSource;
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.interactions.PointerInput;
+import org.openqa.selenium.interactions.Sequence;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+import static com.codeborne.selenide.appium.WebdriverUnwrapper.cast;
+import static java.time.Duration.ofMillis;
+import static java.util.Collections.singletonList;
+
+
+@ParametersAreNonnullByDefault
+public class AppiumScrollTo extends ScrollTo {
+
+  @Override
+  @Nonnull
+  public SelenideElement execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
+    Optional<AppiumDriver> appiumDriverOptional = cast(locator.driver(), AppiumDriver.class);
+    if (!appiumDriverOptional.isPresent()) {
+      return super.execute(proxy, locator, args);
+    }
+    AppiumDriver appiumDriver = appiumDriverOptional.get();
+
+    int currentSwipeCount = 0;
+    String previousPageSource = "";
+
+    while (isElementNotDisplayed(locator)
+      && isNotEndOfPage(appiumDriver, previousPageSource)
+      && isLessThanMaxSwipeCount(currentSwipeCount)) {
+      previousPageSource = appiumDriver.getPageSource();
+      performScroll(appiumDriver);
+      currentSwipeCount++;
+    }
+    return proxy;
+  }
+
+  private boolean isLessThanMaxSwipeCount(int currentSwipeCount) {
+    final int maxSwipeCount = 30;
+    return currentSwipeCount < maxSwipeCount;
+  }
+
+  private boolean isElementNotDisplayed(WebElementSource locator) {
+    try {
+      return !locator.getWebElement().isDisplayed();
```

Comment:
Just a note to use visible attribute not false to continue scrolling in ios. 

For ios, isDisplayed searches entire screen and not just the visible part

@asolntsev